cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/constraint/ConstraintContext.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.schedule.constraint;
+
+import co.cask.cdap.proto.Notification;
+import co.cask.cdap.proto.ProgramRunStatus;
+import co.cask.cdap.proto.RunRecord;
+
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Context object, exposing information that may be useful during checking constraints.
+ */
+public final class ConstraintContext {
```

Comment:
I think we should have a `timestamp` for the context, representing the logical time when this constraint is checked.
The constraint checker may be checking a group of constraints for a pending job and it'll be good for all constraints see the same check time. More importantly, this also make unit-test easy as you don't need to rely on wall clock time, which requires sleep to check in some cases.